### ISSUE TITLE
Can't exit Visual Mode

### ISSUE THREAD

**Issue by andradei** [aioutecism/amVim-for-VSCode#106]
Once I type `V` and enter Visual Mode `ESC` won't exit it. I have to type `c` or `x` or one of those commands, then _undo_ so I can go back to Normal Mode.

__Plugins__
- Go
- Rusty Code
- Preview

__System__
- VS Code 1.2.0
- amVim 1.8.0
- OSX 10.11.5 El Capitan

---

Thanks for the awesome plugin.

**Comment by aioutecism** [comment]
Hi, @andradei 

Do you continue experience this problem after restarting VSCode?
And, what extensions have you installed?

**Comment by chris-dudley** [comment]
I have also experienced the same problem. I have noticed specifically that this occurs only when loading files from a workspace. Editing files without having a workspace/folder open does not seem to cause any problems.

Upon loading a workspace (by opening a new folder, or having VSCode automatically open the last folder you were working on), I see the following errors in VSCode's console in the developer tools:

```
[Extension Host] Activating extension `auiworks.amvim` failed: Cannot read property 'options' of undefined.
[Extension Host] Here is the error stack:  TypeError: Cannot read property 'options' of undefined
    at Function.ActionBlockCursor.on (/home/cdudley/.vscode/extensions/auiworks.amvim-1.8.0/out/src/Actions/BlockCursor.js:8:51)
    at ModeNormal.enter (/home/cdudley/.vscode/extensions/auiworks.amvim-1.8.0/out/src/Modes/Normal.js:188:41)
    at Dispatcher.switchMode (/home/cdudley/.vscode/extensions/auiworks.amvim-1.8.0/out/src/Dispatcher.js:71:26)
    at new Dispatcher (/home/cdudley/.vscode/extensions/auiworks.amvim-1.8.0/out/src/Dispatcher.js:38:14)
    at activate (/home/cdudley/.vscode/extensions/auiworks.amvim-1.8.0/out/src/extension.js:7:18)
    at Function.t._callActivateOptional (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:12:24266)
    at Function.t._callActivate (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:12:24051)
    at /usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:12:23923
    at Object.h [as _notify] (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:5:13179)
    at Object.__dirname.undefined.M.enter (/usr/share/code/resources/app/out/vs/workbench/node/extensionHostProcess.js:5:16341)
```

The error seems to occur [here](https://github.com/aioutecism/amVim-for-VSCode/blob/master/src/Actions/BlockCursor.ts#L10), with `window.activeTextEditor` being undefined. From what I can tell, this may be related to [this issue](https://github.com/Microsoft/vscode/issues/38) in VSCode where `activeTextEditor` is not correctly set during extension activation when loading from a workspace.

Hopefully this information helps.

**Comment by andradei** [comment]
@aioutecism I updated the info above.

**Comment by frederickfogerty** [comment]
I am having this issue too - I cannot reproduce it at the moment, but it just seems to happen later on in a session (i.e. after 10 mins). A restart always fixes it. I will update if I can find a way to reproduce it.

@chris-dudley how do you get to that console view?

**Comment by chris-dudley** [comment]
@frederickfogerty: In the toolbar, Help -> Toggle Developer Tools

**Comment by andradei** [comment]
It happened again (a little before 10 minutes of use). Here's a console output. I don't know how to grab the text in detail, so here goes a picture:

<img width="1150" alt="screen shot 2016-06-14 at 19 12 38" src="https://cloud.githubusercontent.com/assets/2653546/16065044/283ad4a4-3264-11e6-9c67-6f35d813eb09.png">

**Comment by aioutecism** [comment]
This should be fixed in amVim 1.8.1.

**Comment by frederickfogerty** [comment]
Thanks @aioutecism :)
